src/main/scala/geekie/mapred/MapredWorker.scala

```diff
@@ -0,0 +1,83 @@
+package geekie.mapred
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.routing.{ActorRefRoutee, RemoveRoutee}
+import geekie.mapred.MapredWorker._
+
+import scala.collection.TraversableLike
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.reflect.ClassTag
+
+class MapredWorker[A: ClassTag, K: ClassTag, V: ClassTag](mf: A => TraversableOnce[MapredWorker.KeyVal[K, V]],
+                                                          rf: (V, V) => V,
+                                                          ff: Map[K, V] => Unit) extends Actor {
+
+  var acc = Map.empty[K, V]
+
+  def aggregateNewData(kvs: TraversableOnce[KeyVal[K, V]]) = kvs foreach {
+    case KeyVal(k, v) =>
+      acc += (k -> (if (acc contains k) rf(acc(k), v) else v))
```

Comment:
Since this `var acc` is already mutable, what do you think about using `scala.collections.mutable.Map` for performance? I wonder how faster it would be.